src/main/java/util/JavaVersion.java

```diff
@@ -0,0 +1,93 @@
+package util;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Comparator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents Java version
+ */
+public class JavaVersion implements Comparable<JavaVersion> {
+    private static final Logger logger = LogManager.getLogger(JavaVersion.class.getName());
+
+    private static final Pattern JAVA_8_VERSION_PATTERN =
+            Pattern.compile("(\\d+)\\.(\\d+)\\.(\\d+)\\_(\\d+)(-b(\\d+))?");
+
+    public static final String REQUIRED_VERSION = "1.8.0_60";
+
+    private final int discard;
+    private final int major;
+    private final int minor;
+    private final int update;
+    private final int build;
+
+    public JavaVersion(int discard, int major, int minor, int update, int build) {
+        this.discard = discard;
+        this.major = major;
+        this.minor = minor;
+        this.update = update;
+        this.build = build;
+    }
+
+    public static JavaVersion fromString(String javaVersion) {
+        Matcher javaVersionMatcher = JAVA_8_VERSION_PATTERN.matcher(javaVersion);
+
+        if (javaVersionMatcher.find()) {
+            return new JavaVersion(Integer.parseInt(javaVersionMatcher.group(1)),
+                    Integer.parseInt(javaVersionMatcher.group(2)),
+                    Integer.parseInt(javaVersionMatcher.group(3)),
+                    Integer.parseInt(javaVersionMatcher.group(4)),
+                    Integer.parseInt((javaVersionMatcher.group(6) != null ? javaVersionMatcher.group(6) : "0")));
+        } else {
+            logger.error("Java version not valid");
+            return null;
+        }
+    }
+
+    public String toString() {
+        return String.format("%1$d.%2$d.%3$d_%4$d-b%5$d", discard, major, minor, update, build);
+    }
+
+    public int compareTo(JavaVersion other) {
+        if (this.equals(other)) {
+            return 0;
+        }
+
+        return this.discard < other.discard ? -1 :
```

Comment:
Maybe we could define equality (the more specific concept) in terms of Comparable (the more general concept) instead? A bit more DRY. So this would be:

```java
return this.major != other.major ? this.major - other.major :
    this.minor != other.minor ? this.minor - other.minor : 0;
```

and `equals` would be `this.compareTo(other) == 0` (along with null checks, reference equality, etc.).